Lib/test/test_ssl.py

```diff
@@ -2324,15 +2327,22 @@ def run(self):
                         )
                     self.close()
                     self.running = False
-                except OSError:
-                    if self.server.chatty:
-                        handle_error("Test server failure:\n")
-                    self.close()
-                    self.running = False
+                except OSError as err:
+                    if 'peer did not return a certificate' in err.args[1]:
```

Comment:
I wonder why I haven't run into this issue on Linux.

Reply:
The error is coming from Winsock.  Could there be a difference in the way Winsock handles certs and the way linux sockets handle certs that is causing this difference?

Reply:
Most of the tests that are failing are failing in the same way as Linux.  I can tell because there a comments and expected exceptions in multiple places that say something like "sometimes connections fail with TLS 1.3"

The tests are opening a socket doing a handshake and the disconnecting the client before the session tickets are finished sending which results in various disconnect errors.  A real client would send something before closing the session, which appears to allow the session tickets to get sent.  In my mind these are bad tests.  

There are also these issues in the openssl issues about the session ticket protocol: https://github.com/openssl/openssl/issues/7948, https://github.com/openssl/openssl/issues/7967, https://github.com/python-trio/trio/issues/819 which don't really seem resolved.

What is needed to move this PR forward?

Reply:
Looking at the first of those issues, it seems like we should fix our tests to be negotiating the close properly.

Reply:
@tiran Why the OSError case is different on Linux I'm not sure.  On windows the OSError I was trying to handle was SSLError exception (based on OSError).  There is a race condition between the server getting the server getting the PEER_DID_NOT_RETURN_A_CERTIFICATE error and the client receiving the "tvs13 alert ..." error on Windows.  One of them always arrives but the protocol doesn't look like it guarantees enough synchronicity to force one of them to be first.


@zooba  The first 3 tests failures were caused by a bug in openssl so I have removed the try/excepts.  
I have submitted patches for openssl 
https://github.com/openssl/openssl/pull/8590
https://github.com/python/cpython-source-deps/pull/13